packages/jest/src/generators/init/init.ts

```diff
@@ -34,74 +35,89 @@ const schemaDefaults = {
     tree.write(
       `jest.preset.js`,
       `
       const nxPreset = require('@nrwl/jest/preset').default;
-     
+
       module.exports = { ...nxPreset }`
     );
 
     addTestInputs(tree);
   }
-  const rootJestConfig = findRootJestConfig(tree);
-
-  if (options.rootProject && !rootJestConfig) {
-    // we don't want any config to be made because the jest-project generator
-    // will make the config when using rootProject
+  if (options.rootProject) {
+    // we don't want any config to be made because the `jestProjectGenerator`
+    // will copy the template config file
+    return;
+  }
+  const rootJestPath = findRootJestConfig(tree);
+  if (!rootJestPath) {
+    // if there's not root jest config, we will create one and return
+    // this can happen when:
+    // - root jest config was renamed => in which case there is migration needed
+    // - root project didn't have jest setup => again, no migration is needed
+    generateGlobalConfig(tree, options.js);
     return;
   }
+
   const isProjectConfig =
-    rootJestConfig &&
-    tree.exists(rootJestConfig) &&
-    !tree.read(rootJestConfig, 'utf-8').includes('getJestProjects()');
-  if (!options.rootProject && isProjectConfig) {
-    // moving from single project to multi project.
-    // TODO(caleb): this is brittle and needs to be detected better?
-    const nxJson = readJson(tree, 'nx.json');
-    const defaultProject =
-      nxJson.cli?.defaultProjectName || nxJson?.defaultProject;
-
-    const ext = extname(rootJestConfig);
-    const newJestConfigPath = defaultProject
-      ? `jest.${defaultProject}.config${ext}`
-      : `jest.project-config${ext}`;
-
-    tree.rename(rootJestConfig, newJestConfigPath);
-
-    if (defaultProject) {
-      const projectConfig = readProjectConfiguration(tree, defaultProject);
-
-      // TODO(caleb): do we care about a custom target name?
-      if (projectConfig?.targets?.['test']?.options?.jestConfig) {
-        projectConfig.targets['test'].options.jestConfig = newJestConfigPath;
-        updateProjectConfiguration(tree, defaultProject, projectConfig);
+    rootJestPath &&
+    // TODO: (meeroslav) is this condition enough?
+    !tree.read(rootJestPath, 'utf-8').includes('getJestProjects()');
+
+  if (isProjectConfig) {
+    // moving from root project config to monorepo-style config
+    const projects = readWorkspace(tree).projects;
+    const projectNames = Object.keys(projects);
+    const rootProject = projectNames.find(
+      (projecName) => projects[projecName].root === '.'
+    );
+    // root project might have been removed,
+    // if it's missing there's nothing to migrate
+    if (rootProject) {
+      const jestTarget = Object.values(
+        projects[rootProject].targets || {}
+      ).find((t) => t.executor === '@nrwl/jest:jest');
+      // if root project doesn't have jest target, there's nothing to migrate
+      if (jestTarget) {
+        const pathSegments = rootJestPath
+          .split(FILE_EXTENSION_REGEX)
```

Comment:
This regex splits the file path into `path/to/name` and `.ext`

Reply:
removed in favor of just always calling jest.config.app.{ext} based on Jasons feedback